### ISSUE TITLE
Keybinding error; Virtual Mouse on/off

### ISSUE THREAD

**Issue by kgersen** [kgersen/Allegiance#54]
When the virtual mouse is off the keybindings are incorrect. It seems like any "deleted" keymappings aren't actually removed from this key-set and thus can still be used when they virtual mouse is off.

Very odd.

Reported by snufkin and Wizard
http://www.freeallegiance.org/forums/index.php?showtopic=54384
Confirmed by Dorjan

 _Imported from trac ticket #56,  created by Dorjan on 01-12-2010 at 00:26, last modified: 01-08-2011 at 22:43_

**Comment by kgersen** [comment]
there are some TrekKeys that can always be used, including the recent ones i added for mousewheel and xbuttons - is the fix for this to remove all these default "hard coded" key bindings (where it still activates the key even though you don't have it mapped to anything)?

_Trac comment by Imago on 07-19-2010 at 07:33_

**Comment by kgersen** [comment]
if this is the case we'd have to provide a more robust or complete defaultinputmap so that everything is mapped by default.

_Trac comment by Imago on 07-19-2010 at 07:34_

**Comment by kgersen** [comment]
Did some sniffing, I suspect the problem lies around line 610 of trekinput.cpp

"//Imago save special buttons for use outside virtual joystick 8/14/09"

It's mapping those mouse buttons to the hard coded function when they're pressed outside virtual joystick, which removes the existing map somehow.

As far as I can tell, solutions would be to either remove that switch completely and don't worry about mouse button x features unless virtual joystick is on, OR take the function for those buttons from the key map and if they're not mapped don't touch them.

_Trac comment by anonymous on 07-20-2010 at 10:27_

**Comment by kgersen** [comment]
Removed the necessity for a default map, triggered OnTrekKey when the the button is depressed, and worked around the key becoming unassociated with the keyboard mapping. 
The problem was the button state was being tracked with a pointer array, and when the button was pressed outside virtual joystick, the address of the pointer changed instead of the value it pointed to. I fixed it by creating an extra array of booleans for the sole purpose of tracking mouse buttons outside virtual joystick, and setting and checking them independently.

_Trac comment by TurkeyXIII on 11-03-2010 at 08:02_
